### PR TITLE
Support Solidity 0.8.8

### PR DESCRIPTION
**Pull request by feuGeneA**
- [x] Ensure existing tests pass.
- [x] Add tests of new Language Feature, "User Defined Value Type: allows creating a zero cost abstraction over a value type with stricter type requirements."  Ensure that UDVT's can be used as params and return values.

### CODE REVIEW

**Review by fvictorio — APPROVED** [review]

File: .changeset/smooth-turkeys-grab.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"hardhat": patch
+"docs": patch
```
Comment (alcuadrado) [inline .changeset/smooth-turkeys-grab.md:3]: I hate that the link adds docs autmatically. It breaks changesets.
```suggestion
```

Comment (fvictorio) [reply .changeset/smooth-turkeys-grab.md]: Oops, didn't notice that.

File: packages/hardhat-core/test/internal/hardhat-network/stack-traces/test-files/0_8/user-defined-value-types/c.sol [hunk 2]
```diff
@@ -0,0 +1,12 @@
+pragma solidity ^0.8.8;
+
+import "./../../../../../../../console.sol";
```
Comment (fvictorio) [inline packages/hardhat-core/test/internal/hardhat-network/stack-traces/test-files/0_8/user-defined-value-types/c.sol:3]: You can remove this now (and also the import in the `test.json`).

Comment (feuGeneA) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test-files/0_8/user-defined-value-types/c.sol]: Addressed in https://github.com/nomiclabs/hardhat/pull/2017/commits/ba4bfc18efd3c0fba58a98b9821fd42db3ddd9f7
